### PR TITLE
pg-cdc: support numeric data types without scale and precision

### LINKED ISSUE #9269: Error on Numeric Datatype for CREATE VIEWS
A table containing a numeric datatype fails to run <code>create views from source mz_source</code>. Code example: <pre><code>materialize=> create views from source mz_source ("wallet_balances");
ERROR:  precision for type numeric must be within [1-39], have 65535</code></pre>
There is a PR out to fix this: https://github.com/MaterializeInc/materialize/pull/9268

### PR DESCRIPTION
**Pull request by petrosagg**
### Motivation

We would previously error out when encountering an upstream table with numeric columns that had no scale and precision specified because we failed to test the type modifier for validity.

That led to wrap-around behaviour of the computed values that were way above our limit of 39 decimal digits.

Fixes #9269

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR adds a release note for any
  [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

File: doc/user/content/release-notes.md [hunk 1]
```diff
@@ -103,6 +103,11 @@ List new features before bug fixes.
 
 {{% version-header v0.10.1 %}}
 
+- Correctly autogenerate views from Postgres sources during [`CREATE
+  VIEWS`](/sql/create-source/postgres/#creating-views) when the upstream table
+  contains numeric columns with no specified scale and precision. {{% gh 9268
+  %}}
+
```
Comment (andrioni) [inline doc/user/content/release-notes.md:110]: What happens if we ingest a numeric value that doesn't fit in our numeric type?

Comment (petrosagg) [reply doc/user/content/release-notes.md:110]: The source will fail with an error message like this one:

```
materialize=> SELECT * FROM foo;
ERROR:  "1234567890123456789012345678901234567890" is out of range for type numeric: exceeds maximum precision 39
```

File: src/postgres-util/src/lib.rs [hunk 2]
```diff
@@ -234,16 +246,19 @@ pub async fn publication_info(
...
                         let modifier: i32 = row.get("modifier");
-                        // https://github.com/postgres/postgres/blob/REL_13_3/src/backend/utils/adt/numeric.c#L978-L983
-                        let tmp_mod = modifier - 4;
-                        let scale = (tmp_mod & 0xffff) as u16;
-                        let precision = ((tmp_mod >> 16) & 0xffff) as u16;
+                        // https://github.com/postgres/postgres/blob/REL_13_3/src/backend/utils/adt/numeric.c#L967-L983
+                        let typ_mod = if modifier < 4 {
+                            None
+                        } else {
+                            let tmp_mod = modifier - 4;
```
Comment (quodlibetor) [inline src/postgres-util/src/lib.rs:253]: IMO it's worth us adding a `const PG_HDRSZ: i32 = 4` (or `PG_HEADER_SIZE`) to mirror pg's `VARHDRSZ`.

File: src/postgres-util/src/lib.rs [hunk 3]
```diff
@@ -23,10 +23,15 @@ use sql_parser::ast::display::{AstDisplay, AstFormatter};
...
+    precision: u16,
+    scale: u16,
+}
+
 pub enum PgScalarType {
     Simple(PgType),
-    Numeric { precision: u16, scale: u16 },
-    NumericArray { precision: u16, scale: u16 },
+    Numeric(Option<PgNumericMod>),
+    NumericArray(Option<PgNumericMod>),
```
[the remark this answers is not in the document]
Comment (petrosagg) [reply src/postgres-util/src/lib.rs:34]: done

File: src/postgres-util/src/lib.rs (hunk 2 above)
Comment (petrosagg) [reply src/postgres-util/src/lib.rs]: good idea! done

**Pull request merged**